src/callbacks_step/time_series_dg2d.jl

```diff
@@ -68,6 +70,52 @@ function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh,
     return element_ids
 end
 
+# Elements on an `UnstructuredMesh2D` are possibly curved. Assume that each
+# element is convex, i.e., all interior angles are less than 180 degrees.
+# We use the barycenter of each element to determine if a given coordinate (x,y)
+# lies within said element. The shortest distance between the point and all the
+# barycenters "wins".
+function get_elements_by_coordinates!(element_ids, coordinates,
+                                      mesh::UnstructuredMesh2D,
+                                      dg, cache)
+    if length(element_ids) != size(coordinates, 2)
+        throw(DimensionMismatch("storage length for element ids does not match the number of coordinates"))
+    end
+
+    # Reset element ids - 0 indicates "not (yet) found"
+    element_ids .= 0
+
+    # Compute and save the barycentric coordinate on each element
+    bary_centers = zeros(eltype(mesh.corners), 2, mesh.n_elements)
+    distances = zeros(eltype(mesh.corners), mesh.n_elements)
+    calc_bary_centers!(bary_centers, dg, cache)
+
+    # Iterate over coordinates
+    for index in 1:length(element_ids)
+        # Compute the distance between the current point and all the bary centers.
+        for element in eachelement(dg, cache)
+            distances[element] = norm(coordinates[:, index] .- bary_centers[:, element])
+        end
+
+        # Locate the minimal distance as this gives the element containing the point
+        element_ids[index] = argmin(distances)
+    end
+
+    return element_ids
+end
+
+# Use the available `node_coordinates` on each element to compute and save the barycenter.
+@inline function calc_bary_centers!(bary_centers, dg, cache)
+    n = nnodes(dg)
+    for element in eachelement(dg, cache)
+        bary_centers[1, element] = sum(cache.elements.node_coordinates[1, :, :,
+                                                                       element]) / n^2
+        bary_centers[2, element] = sum(cache.elements.node_coordinates[2, :, :,
+                                                                       element]) / n^2
+    end
+    return nothing
+end
```

Comment:
```suggestion
@inline function calc_bary_centers!(bary_centers, dg, cache)
    n = nnodes(dg)
    @views for element in eachelement(dg, cache)
        bary_centers[1, element] = sum(cache.elements.node_coordinates[1, :, :,
                                                                       element]) / n^2
        bary_centers[2, element] = sum(cache.elements.node_coordinates[2, :, :,
                                                                       element]) / n^2
    end
    return nothing
end
```

To avoid allocations